sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -722,6 +722,9 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
           REPARTITION_BY_NUM
         }
         exchange.ShuffleExchangeExec(r.partitioning, planLater(r.child), shuffleOrigin) :: Nil
+      case r: logical.AdaptiveRepartition =>
+        exchange.ShuffleExchangeExec(
+          r.partitioning, planLater(r.child), ADAPTIVE_REPARTITION) :: Nil
```

Comment:
Do you mean use local shuffle reader to improve this case?
```sql
CREATE TABLE t1 USING parquet PARTITIONED BY (p2)
AS
(SELECT id, id % 1000 AS p1, id AS p2 FROM range(3100000) distribute by p2)
```

Reply:
`distribute by ` has its semantics so this PR add a new operator and hint which doesn't guarantee the output paritionging. And for your case, we can do like this:
```
CREATE TABLE t1 USING parquet PARTITIONED BY (p2)
AS
(SELECT /*+ ADAPTIVE_REPARTITION(p2) */ id, id % 1000 AS p1, id AS p2 FROM range(3100000))
```